app/src/androidTest/java/com/github/ybecker/epforuml/database/MockDatabaseTest.kt

```diff
@@ -301,4 +300,23 @@ class MockDatabaseTest {
         }.join()
     }
 
+    @Test
+    fun setUserPresenceAddsConnection() {
+        db.addUser("0", "test", "testEmail").join()
+        db.setUserPresence("0")
+        db.getUserById("0").thenAccept {
+            assertTrue(it!!.connections.size == 1)
+        }
+    }
+
```

Comment:
It would be good to test other scenarios, such as what happens if you try to add the same connection twice, or if you try to remove a connection that doesn't exist.